FirmataParser.cpp

```diff
@@ -163,8 +166,8 @@ void FirmataParser::parse(uint8_t inputData)
         systemReset();
         break;
       case REPORT_VERSION:
-        if (currentReportVersionCallback)
-          (*currentReportVersionCallback)(currentReportVersionCallbackContext);
+        waitForData = 2; // two data bytes needed
```

Comment:
I think there is some general confusion here because `REPORT_VERSION` is misleading. Basically a client should never report its version to the host/firmware. A client can only query the firmware and protocol versions from the host/firmware.